### PR TITLE
ACIC2016 dataloader & FlexTENet

### PR DESCRIPTION
**Pull request by bcebere**
This PR adds a new dataloader for the ACIC2016 dataset and the PyTorch version of FlexTENet.

 - [x] Experiments updates
 - [x] Unit tests

### CODE REVIEW

File: catenets/datasets/dataset_acic2016.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+    test_potential_y: array or pd.DataFrame
+        Potential outcomes in testing data.
+    """
+    arch = data_path / "data_cf_all.tar.gz"
+
+    download_if_needed(
+        arch, file_id=FILE_ID, unarchive=True, unarchive_folder=data_path
+    )
+
+    csv = data_path / "data_cf_all/x.csv"
```
Comment (AliciaCurth) [inline catenets/datasets/dataset_acic2016.py:228]: this uses the wrong x_file for this simulation; I think I told you that there's a local preprocessed file (X_trans) that I used for this (which is preprocessed using an old R-script from the original acic2016, that's why it would be easier to just host the pre-transformed file somewhere I think).

Comment (AliciaCurth) [reply catenets/datasets/dataset_acic2016.py:228]: That being said, there is actually a second set of experiments that I have run on the original acic data + outcomes (so it uses x.csv as well as the other files in the folders in the archive) [these experiments are not in this repo yet]; so it wouldnt hurt to have an option to choose whether to use x_trans.csv or x.csv :)

File: catenets/models/torch/snet.py [hunk 2]
```diff
@@ -310,9 +310,22 @@ def _get_absolute_rowsums(mat: torch) -> torch.Tensor:
...
         ) -> torch.Tensor:
-            return torch.linalg.norm(torch.dot(params_0.T, params_1), "fro") ** 2
+            if normalize:
+                params_0 = params_0 / torch.linalg.norm(params_0, dim=0)
+                params_1 = params_1 / torch.linalg.norm(params_1, dim=0)
+
+            x_min = min(params_0.shape[0], params_1.shape[0])
+            y_min = min(params_0.shape[1], params_1.shape[1])
+            return (
+                torch.linalg.norm(
```
Comment (AliciaCurth) [inline catenets/models/torch/snet.py:322]: I think this should stay  ``torch.linalg.norm(torch.dot(....), "fro")`` instead of the CosineSimilarity() functions. If not, the normalization doesn't make sense (cosine similarity normalizes internally, if I see this correctly). We can rename the function (get_cos_reg) to something else (e.g. get_fro_reg ; fro for frobenius) if you think its misleading because of the cosine in the name.

File: catenets/datasets/dataset_acic2016.py (hunk 1 above)
Comment (bcebere) [reply catenets/datasets/dataset_acic2016.py:228]: The GDrive share you've sent me doesn't include x_trans.csv, only x.csv.
